res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,425 @@
+            size: NumarkPartyMix.autoLoopSizes[i],
+            inKey: "beatloop_" + NumarkPartyMix.autoLoopSizes[i] + "_toggle",
+            outKey: "beatloop_" + NumarkPartyMix.autoLoopSizes[i] + "_enabled",
+            outConnect: false
+        });
+    }
+};
+NumarkPartyMix.ModeLoop.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.ModeSampler = function(deckNumber) {
+    components.ComponentContainer.call(this);
+    this.control = NumarkPartyMix.PadModeControls.SAMPLER;
+    var sampleoffset = (deckNumber - 1) * 4;
+    this.connections = new components.ComponentContainer();
+    for (var i = 0; i < 4; i++) {
+        this.connections[i] = new components.SamplerButton({
+            midi: [0x93 + deckNumber, 0x14 + i],
+            number: 1 + i + sampleoffset,
+            outConnect: false,
+            unshift: null,
+            outKey: "play_indicator",
+            input: function(channel, control, value, status, _group) {
+                if (this.isPress(channel, control, value, status)) {
+                    if (engine.getValue(this.group, "track_loaded") === 0) {
+                        engine.setValue(this.group, "LoadSelectedTrack", 1);
+                    } else {
+                        if (engine.getValue(this.group, "play") === 1) {
+                            engine.setValue(this.group, "start_stop", 1);
+                        } else {
+                            engine.setValue(this.group, "start_play", 1);
+                        }
+                    }
+                }
+            }
+        });
+    }
+};
+NumarkPartyMix.ModeSampler.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.ModeEFX = function(deckNumber) {
+    components.ComponentContainer.call(this);
+
+    this.control = NumarkPartyMix.PadModeControls.EFX;
+
+    var fx = [
+        "[EffectRack1_EffectUnit" + deckNumber + "_Effect1]",
+        "[EffectRack1_EffectUnit" + deckNumber + "_Effect2]",
+        "[EffectRack1_EffectUnit" + deckNumber + "_Effect3]"
+    ];
+
+    this.connections = new components.ComponentContainer();
+    var p = this.connections;
+
+    fx.forEach(function(fx, i) {
+        p[i] = new components.Button({
+            midi: [0x93 + deckNumber, 0x14 + i],
+            group: fx,
+            inKey: "enabled",
+            outKey: "enabled",
+            outConnect: false
+        });
+    });
+    p[3] = new components.Button({
+        midi: [0x93 + deckNumber, 0x17],
+        type: components.Button.prototype.types.toggle,
+        group: "[EffectRack1_EffectUnit" + deckNumber + "]",
+        inKey: "mix_mode",
+        outKey: "mix_mode",
+        output: function() {
+            if (engine.getParameter("[EffectRack1_EffectUnit" + deckNumber + "]", "mix_mode") === 0) {
+                midi.sendShortMsg(0x93 + deckNumber, 0x17, 0x7F);
+            } else {
+                midi.sendShortMsg(0x83 + deckNumber, 0x17, 0x01);
+            }
```

Comment:
to only switch the effect on during button press

Reply:
Can you elaborate? I don't understand yet.

Reply:
I'm sorry I still don't understand. If I understand correctly, when a pad is supposed to be off, then its dimly lit, correct? Why do we make an exception here then and shut off the pad completely? And why exactly is the outkey useless here?

Reply:
what is your suggestion?

Reply:
Just keep it the same as the other buttons
```js
    p[3] = new components.Button({
        midi: [0x93 + deckNumber, 0x17],
        type: components.Button.prototype.types.toggle,
        group: "[EffectRack1_EffectUnit" + deckNumber + "]",
        key: "mix_mode",
        outConnect: false,
    });

Reply:
I realize I did this to swap the LED so that it is on with the default (Dry/Wet Mode). this is helpful to distinguish it from the other three buttons.

Reply:
it is necessary to indicate that the 4th button is different from the other 3. i use outValueScale to achieve this.

Reply:
Sure, thats the intended use for that function.